src/PIL/ImageColor.py

```diff
@@ -87,13 +87,13 @@ def getrgb(color):
             int((int(m.group(3)) * 255) / 100.0 + 0.5)
             )
 
-    m = re.match(r"hsl\(\s*(\d+)\s*,\s*(\d+)%\s*,\s*(\d+)%\s*\)$", color)
+    m = re.match(r"hsl\(\s*(\d+(\.\d*)?)\s*,\s*(\d+(\.\d*)?)%\s*,\s*(\d+(\.\d*)?)%\s*\)$", color)
```

Comment:
Do we need brackets around these new decimal matchers?

If they're not being used in a `m.group(X)`, they can be omitted to make the regex more readable (readable in relative terms!).

Reply:
Okay, I've updated the commit, see what you think.